### ISSUE TITLE
[BUG] 10.6.0 regresses stdout pollution on npx/npm exec

### ISSUE THREAD

**Issue by lorenzogrv** [npm/cli#7435]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

npm pollutes stdtout when using npx/npm exec

Old behavior:

```bash
$ npm --version
10.2.4
$ npm exec -- /bin/bash -c 'echo what?'
what?
```

```bash
$ npm install -g npm@latest

removed 43 packages, and changed 74 packages in 710ms

24 packages are looking for funding
  run `npm fund` for details
```
New behavior

```bash
$ npm --version
10.6.0
$ npm exec -- /bin/bash -c 'echo what?'

> @grupomarea/roo@0.1.0 npx
> /bin/bash -c echo what?

what?
```

> This seems similar to https://github.com/npm/cli/issues/7354

### Expected Behavior

```bash
$ npm --version
10.6.0
$ npm exec -- /bin/bash -c 'echo what?'
what?
```

### Steps To Reproduce

1. Follow the steps above

### Environment

- system

```bash
$ cat /etc/lsb-release 
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=21.1
DISTRIB_CODENAME=vera
DISTRIB_DESCRIPTION="Linux Mint 21.1 Vera"
$ uname -i
x86_64
```
- npm config:

```ini
; node version = v20.11.1
; npm version = 10.6.0
```

**Comment by lorenzogrv** [comment]
For a sake of completeness, this does not happen on `10.5.0`

```bash
$ npm install -g npm@10.5
# ...
$ npm exec -- /bin/bash -c 'echo what?'
what?
```

PS: Why `v10.6.0` is tagged `@latest` when [current docs changelog](https://docs.npmjs.com/cli/v10/using-npm/changelog) states `v10.5.2` is latest CLI?

**Comment by lorenzogrv** [comment]
@wraithgar seems also related to https://github.com/npm/statusboard/issues/829

**Comment by milaninfy** [comment]
This is duplicate of #7419 and it is fixed in this PR #7421.

**Comment by lorenzogrv** [comment]
> This is duplicate of #7419 and it is fixed in this PR #7421.

Ok, sorry the noise

**Comment by milaninfy** [comment]
No Problem. Thank you for testing with latest npm.
